caas/kubernetes/provider/admissionregistration.go

```diff
@@ -65,6 +66,22 @@ func (k *kubernetesClient) ensureMutatingWebhookConfigurations(
 				ObjectMeta: obj,
 				Webhooks:   toMutatingWebhookV1beta1(v.Webhooks),
 			})
+			if errors.Cause(err) == errVersionNotSupported {
```

Comment:
We do something similar already for the networking beta vs v1 apis; there we look at the k8s api version to know up front if we need to do the conversion, rather than going to the expense of trying it and reacting to an error. We also handle the case where a new charm with v1 is deployed to an older cluster with beta api. Although that becomes less of an issue over time.
Also, the code to do the conversion is in the specs package with all the other yaml parsing functionality.